program/rust/src/rust_oracle.rs

```diff
@@ -10,10 +33,61 @@ pub fn update_price(
     instruction_data: &[u8],
     input: *mut u8,
 ) -> OracleResult {
-    //For now, we did not change the behavior of this. this is just to show the proposed structure
-    // of the program
-    c_entrypoint_wrapper(input)
+    let account_len = accounts[1].try_data_len()?;
```

Comment:
I feel like this would be better in function `has_been_resized` or something like that

Reply:
You can error inside the function if it doesn't match one of the two allowed sizes.